elastic/kibana
# [Controls] Clear range/time slider selections when field changes

Closes https://github.com/elastic/kibana/issues/129638

## Summary

Before this change, old selections would remain when a control's field was edited for both **range** and **time** slider controls - this would, more often than not, result in an invalid selection for the new control. Now, if the field is different when the control editor is saved, the previous selections are cleared for all three control types.

### How to test
For both range and time slider controls, you should:
- Test that changing the field clears selections:
  1. Create a control of that specific type
  2. Make some sort of selection with that control
  3. Edit the control and change the field that is being referenced
  4. Save your new control
  5. Notice that the old selections are gone

- Test that **only** field changes clear selections:
  1. Create a control of that specific type
  2. Make some sort of selection with that control
  3. Edit the control and change the size and/or the label and/or some other setting that *is not* the field
  4. Save your new control
  5. Notice that the old selections **remain**

### Videos

**Changing field clears selection:**

https://user-images.githubusercontent.com/8698078/164111572-9a28aa2c-e212-4c7c-bfb7-c3a827d2c4a3.mp4

**Changing anything else *does not* clear selection:**

https://user-images.githubusercontent.com/8698078/164111577-7973548c-71f8-4119-bf8b-927347a3422b.mp4

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
